com.unity.shadergraph/Editor/GraphUI/Importers/ShaderGraphImporterEditor.cs

```diff
@@ -59,17 +63,21 @@ public override void OnDisable()
         public static bool OnOpenShaderGraph(int instanceID, int line)
         {
             string path = AssetDatabase.GetAssetPath(instanceID);
-            var assetModel = AssetDatabase.LoadAssetAtPath(path, typeof(ShaderGraphAssetModel)) as ShaderGraphAssetModel;
-            if(assetModel == null)
-            {
-                return false;
-            }
+            var assetModel = ShaderGraphAsset.HandleLoad(path);
             return ShowWindow(path, assetModel);
         }
 
         private static bool ShowWindow(string path, ShaderGraphAssetModel model)
         {
-            var shaderGraphEditorWindow = GraphViewEditorWindow.FindOrCreateGraphWindow<ShaderGraphEditorWindow>();
+            // Prevents the same graph asset from being opened in two separate editor windows
+            var existingEditorWindows = (ShaderGraphEditorWindow[])Resources.FindObjectsOfTypeAll(typeof(ShaderGraphEditorWindow));
+            foreach (var existingEditorWindow in existingEditorWindows)
+            {
+                if (existingEditorWindow.GraphTool.ToolState.AssetModel == model)
```

Comment:
This line causes an error for me because it is a reference comparison. What is the right thing here?

```
/Users/brett.groshong/github.com/Unity-Technologies/Graphics/com.unity.shadergraph/Editor/GraphUI/Importers/ShaderGraphImporterEditor.cs(73,21): error CS0252: Possible unintended reference comparison; to get a value comparison, cast the left hand side to type 'Object'
```

Reply:
hmm interesting, I don't see that, I can make a change real quick, I want to see if those are the same asset models so I can use `Object.ReferenceEquals`